exercise_3/src/Course.cpp

```diff
@@ -0,0 +1,18 @@
+#include <sstream>
+#include <string>
+#include "Course.h"
+
+Course::Course(int &id, std::string &name, int &teacher_id): _name{name}, _teacher_id{teacher_id} {
+	this -> _id = id;
```

Comment:
In general, "this->" is not needed.

Reply:
I used "this -> " because "_id" is not the member of class Course. It is in protected members of IRecord (base class). As IRecord is interface (according task) I didn't implement it's constructor. That's why I couldn't use _id{id} or IRecord(id) on initializtion list.
The same is about "_name" in Student and Teacher classes.

Reply:
OK, I haven't notice it. 
You may create 
IRecord::IRecord(int id){
}

And then:

Course::Coures(...): IRecord{_id}.


In general, if it is possible (and make sense), we initialize arguments on initialization list. Not in constructor body.

Reply:
Thanks for helping.

I made all constructors arguments "const" and initialized all arguments on initialization list.